### PR TITLE
refactor: remove legacy echo experiment post/patch, trial and trial metrics get endpoints.

### PR DESCRIPTION
**Pull request by ioga**
## Description

External: see release note.
Internal: also get rid of legacy `CreateExperimentParams` struct. it used to be echo query parameters and the new api endpoint shimmied it.

## Test Plan

N/A

## Commentary (optional)

I tripped over the old APIs while doing something else.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/release-notes/remove-legacy-apis.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+:orphan:
+
+**Removed Features**
+
+-  API: Legacy APIs for trial details and trial metrics, which were deprecated in 0.19.2, have now
+   been removed.
+-  API: Legacy APIs for experiment creation and updates, which were deprecated in 0.19.10, have now
+   been removed.
```
Comment (tara-det-ai) [inline docs/release-notes/remove-legacy-apis.rst:8]: LGTM

**Review by stoksc — APPROVED** [review]
lgtm, though it conflicts with https://github.com/determined-ai/determined/pull/6721

maybe you can land this and @nrajanee can rebase the refactor onto main afterwards

**Review by hamidzr — APPROVED** [review]
lgtm. I'm surprised the cli isn't using any of these.

**Comment by ioga** [comment]
oh no, we still use patchExperiment to update the experiment priority and slots from the CLI 😔

**Comment by ioga** [comment]
docs and GHA failures are known issues.

**Pull request merged**